src/main/java/com/google/api/codegen/viewmodel/StaticLangApiMessageView.java

```diff
@@ -75,11 +74,15 @@ public boolean hasRequestBody() {
   public abstract boolean hasRequiredProperties();
 
   // List of ResourceNames that this message contains.
-  public abstract List<RequestObjectParamView> resourceNames();
+  @Nullable
+  public abstract RequestObjectParamView resourceName();
```

Comment:
I'm a little confused about this. I can see how a request message would have a primary resource name, but couldn't there be additional fields that also have resource names?

Reply:
I haven't yet seen multiple resource names for a given Message object in a discovery document, so I don't know what that would look like.